pkg/fanal/image/registry/ecr/ecr.go

```diff
@@ -46,11 +47,34 @@ func (e *ECR) CheckOptions(domain string, option types.RegistryOptions) error {
 		return err
 	}
 
+	// override region with the value from the repository domain
+	cfg.Region = *region
+
 	svc := ecr.NewFromConfig(cfg)
 	e.Client = svc
 	return nil
 }
 
+// Endpoints take the form
+// <registry-id>.dkr.ecr.<region>.amazonaws.com
+// <registry-id>.dkr.ecr-fips.<region>.amazonaws.com
+// <registry-id>.dkr.ecr.<region>.amazonaws.com.cn
+// <registry-id>.dkr.ecr.<region>.sc2s.sgov.gov
+// <registry-id>.dkr.ecr.<region>.c2s.ic.gov
+// see
+// - https://docs.aws.amazon.com/general/latest/gr/ecr.html
+// - https://docs.amazonaws.cn/en_us/aws/latest/userguide/endpoints-arns.html
+// - /usr/local/aws-cli/awscli/botocore/data/endpoints.json
+var ecrEndpointMatch = regexp.MustCompile(`^[^.]+\.dkr\.ecr(?:-fips)?\.([^.]+)\.(?:amazonaws\.com(?:\.cn)?|sc2s\.sgov\.gov|c2s\.ic\.gov)$`)
+
+func determineRegion(domain string) *string {
```

Comment:
Do we need pointer here?

Reply:
I choose to use a pointer to distinguish between not matching the regexp and returning a match, but I can return a boolean instead if that's your preference.

Reply:
You can simply compare with empty string.
 ```
...
region := determineRegion(domain)
	if region == "" {
		return xerrors.Errorf("ECR : %w", types.InvalidURLPattern)
	}
...
func determineRegion(domain string) string {
	matches := ecrEndpointMatch.FindStringSubmatch(domain)
	if matches != nil {
		return matches[1]
	}
	return ""
}
```

Reply:
Technically yes, but not something I'm a fan of to be honest.  I prefer to be explicit.

Reply:
Typically we use comparison logic with an empty string.
I think it's best to stick to one code style.
It may also make this part of the code difficult to maintain.
Because over time, maintainer must remember why we are using a pointer here.